src/Core/Settings.h

```diff
@@ -800,6 +800,8 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Int64, record_consume_timeout_ms, 100, "Timeout of consuming record", 0) \
     M(Int64, kafka_fetch_wait_max_ms, 100, "When reading from Kafka, max wait time", 0) \
     M(Int64, kafka_fetch_max_bytes, 52428800, "When reading from Kafka, max bytes to fetch per read", 0) \
+    M(UInt64, kafka_max_message_size, 1024 * 1024, "When writing to Kafka, the max size of a message in bytes. This setting only works when a row based output format is used. It works together with `kafka_max_message_rows`, when one of these limits is reached, a message will be created.", 0) \
+    M(UInt64, kafka_max_message_rows, DEFAULT_INSERT_BLOCK_SIZE, "When writing to Kafka, the max number of rows of data can be written into one message. This setting only works when a row based output format is used. It works together with `kafka_max_message_size`, when one of these limits is reached, a message will be created.", 0) \
```

Comment:
Can we just use the max_insert_block_size and treat kafka_max_message_size an additional enhancement ?

Reply:
We can. The reason why I introduced this `kafka_max_message_rows` is to keep the settings consistent. Compare `settings kafka_max_message_size = 100, kafka_max_message_rows = 2` and `settings kafka_max_message_size = 100, max_insert_block_size = 2`, the former one looks more reasonable to me.

Let me know what you think.

Reply:
let's use `max_insert_block_size` because users can use this settings for all scenarios and also for kafka ingest. If max_insert_block_size doesn't work for kafka ingest, users got surprised as well.